File: api/Models/Item.cs
namespace SecondBrain.Api.Models;

public class Item
{
    public int Id { get; set; }
    public string ItemType { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Content { get; set; }
    public string? Url { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class CreateItemRequest
{
    public string ItemType { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Content { get; set; }
    public string? Url { get; set; }
}

File: frontend/src/app/services/item.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface Item {
  id: number;
  itemType: string;
  title: string;
  content: string | null;
  url: string | null;
  createdAt: string;
}

export interface CreateItemRequest {
  itemType: string;
  title: string;
  content?: string | null;
  url?: string | null;
}

@Injectable({ providedIn: 'root' })
export class ItemService {
  private readonly apiUrl = `${environment.apiUrl}/items`;

  constructor(private http: HttpClient) {}

  getItems(): Observable<Item[]> {
    return this.http.get<Item[]>(this.apiUrl);
  }

  createItem(request: CreateItemRequest): Observable<Item> {
    return this.http.post<Item>(this.apiUrl, request);
  }
}
